src/vs/workbench/api/browser/mainThreadInlineChat.ts

```diff
@@ -10,11 +10,13 @@ import { reviveWorkspaceEditDto } from 'vs/workbench/api/browser/mainThreadBulkE
 import { ExtHostContext, ExtHostInlineChatShape, MainContext, MainThreadInlineChatShape as MainThreadInlineChatShape } from 'vs/workbench/api/common/extHost.protocol';
 import { IExtHostContext, extHostNamedCustomer } from 'vs/workbench/services/extensions/common/extHostCustomers';
 import { IProgress } from 'vs/platform/progress/common/progress';
+import { Emitter } from 'vs/base/common/event';
 
 @extHostNamedCustomer(MainContext.MainThreadInlineChat)
 export class MainThreadInlineChat implements MainThreadInlineChatShape {
 
 	private readonly _registrations = new DisposableMap<number>();
+	private readonly _changeEvents = new DisposableMap<number, Emitter<void>>();
```

Comment:
Instead of two maps that need to be kept in sync you should use `_registrations` and store a new `Entry` object that is the provider and its emitters